go-controller/pkg/ovn/master.go

```diff
@@ -848,6 +868,28 @@ func (oc *Controller) syncNodes(nodes []interface{}) {
 	} else {
 		subnetAttr = "subnet"
 	}
+
+	//ChassisMap is a map of hostname (nodeName) to chassis name
+	chassisMap := make(map[string]string)
+
+	chassisData, stderr, err := util.RunOVNSbctl("--data=bare", "--no-heading",
+		"--columns=name,hostname", "list", "Chassis")
+	if err != nil {
+		klog.Errorf("Failed to get chassis list: stderr: %q, error: %v",
+			stderr, err)
+		return
+	}
+	for _, chassis := range strings.Split(chassisData, "\n\n") {
```

Comment:
This is pretty ugly imo. Can we get it as json and unmarshal it directly into a map struct?

[root@ovn-control-plane ~]# ovn-sbctl  -f json --columns=name,hostname list chassis 
{"data":[["64da027a-1e35-42f8-af03-59a1f4903c7d","ovn-control-plane"]],"headings":["name","hostname"]}

Reply:
Following existing parsing. But I agree. I will change this to use json unmarshalling.

Reply:
@abhat the chassis ID is already present as a node annotation.. "k8s.ovn.org/node-chassis-id"... see `parseNodeChassisID()`.. you don't need to make any `ovn-sbctl` call

Reply:
@girishmg I'd rather use the node name than the chassis ID. We are guaranteed to get a hostname (because we have it from the Kube API via the node name, and we have it from SBDB because it's in the chassis record). We are not guaranteed to have the chassis ID annotation on the node.

Reply:
@dcbw good point. However, I was trying to avoid `ovn-sbctl` calls from the Master as much as possible as I have seen things timing out at scale. But again, like I said before, calling chassis-del from deleteNode() will not work since we are not guaranteed that ovn-controller process is already killed.